dev/create-release/release-build.sh

```diff
@@ -179,7 +178,7 @@ if [[ "$1" == "package" ]]; then
     rm -r spark-$SPARK_VERSION/licenses-binary
   fi
 
-  tar cvzf spark-$SPARK_VERSION.tgz spark-$SPARK_VERSION
+  tar cvzf spark-$SPARK_VERSION.tgz --exclude spark-$SPARK_VERSION/.git spark-$SPARK_VERSION
```

Comment:
hmm, when we build binary dist later, don't we run `build/spark-build-info` to prepare correct `spark-version-info.properties`?

Reply:
If we do not need ro run `build/spark-build-info` later, how to prepare the correct `spark-version-info.properties` to show the correct information in `bin/spark-submit --version`?

Reply:
I think it's supposed to be ran for both SBT and Maven builds:

```
core/pom.xml:                  <arg value="${project.basedir}/../build/spark-build-info"/>
project/SparkBuild.scala:      val buildScript = baseDirectory.value + "/../build/spark-build-info"
```

I think @viirya meant we might better consider fixing somewhere around `build/spark-build-info` script.

Reply:
I see. `build/spark-build-info` is called from one of two builds and on docker or host.

The straightforward way seems to always keep `.git`. Another possible way may keep this information somewhere and refer to this later.